src/compile/data/impute.ts

```diff
@@ -63,47 +63,44 @@ export class ImputeNode extends DataFlowNode {
     return null;
   }
 
-  public assemble(): [VgImputeTransform] | [VgImputeTransform, VgWindowTransform, VgFormulaTransform] {
-    const {impute, key, keyvals, method, groupby, value, frame} = this.transform;
-    if (method !== 'value' && frame !== undefined) {
+  public assemble(): [VgImputeTransform, VgWindowTransform | VgFormulaTransform, VgFormulaTransform] {
+    const {impute, key, keyvals, method, groupby, value, frame=[null,null]} = this.transform;
+
       const initialImpute: VgImputeTransform = {
         type: 'impute',
         field: impute,
         key,
         ...(keyvals ? {keyvals: isImputeSequence(keyvals) ? this.processSequence(keyvals) : keyvals}: {}),
         method: 'value',
         ...(groupby ? {groupby}: {}),
-        value: undefined
-      };
-
-      const deriveNewField: VgWindowTransform = {
-        type: 'window',
-        as: ['derived_field'],
-        ops: [method],
-        fields: [impute],
-        frame,
-        ignorePeers: false,
-        ...(groupby ? {groupby}: {})
+        value: null
       };
 
+    let deriveNewField: VgWindowTransform | VgFormulaTransform;
+      if (method && method !== 'value') {
+        deriveNewField = {
+          type: 'window',
+          as: ['derived_field'],
```

Comment:
Instead of naming this as a fixed "derived field", can we name it with something more specific to impute?

e.g., `imputed_${field}_value`